core/trino-main/src/main/java/io/trino/operator/OperatorStats.java

```diff
@@ -451,6 +469,8 @@ public OperatorStats add(Iterable<OperatorStats> operators)
         long outputPositions = this.outputPositions;
 
         long dynamicFilterSplitsProcessed = this.dynamicFilterSplitsProcessed;
+        ImmutableList.Builder<Map<String, Metric>> metricsMaps = ImmutableList.builder();
```

Comment:
Do we need a list ?
Could we merge the maps one at a time like the other metrics here ?

Reply:
Sounds good, changed to merge the metrics on at a time.